### PR TITLE
WKT DGGS literal

### PR DESCRIPTION
**Pull request by nicholascar**
I've pretty much followed suite with the WKT literal and @situx 's PR for the GeoJSON literal.

I think that we can have a special (subClassOf) datatype property for WKT DGGS literals that is very similar to a WKT literal but cannot be exactly that as DGGS coordiantes don't allow for valid WKT values.

We can continue to use the `asWKT` property and don't need another one: since the `wktDGGSLiteral` is a `subClassOf` `wktLiteral` this is legal in the ontology.

I may issue other literal properties for DGGS also, e.g. a GeoJSON-like property (again, can't be actual GeoJSON as that's locked to WGS84).

### CODE REVIEW

**Comment by nicholascar** [comment]
I am building a WKT DGGS validator here: https://github.com/surroundaustralia/dggsgv. We (DGGS Testbed 16 & DGGS DWG) should be done with the specification of the WKT DGGS literal format shortly and then we will be able to put that through the DWG for OGC publication.

Even though the WKT DGGS formats might not be formal just yet, I think that this spec, GeoSPARQL 1.1, can just "hand off" to that and need not contain details of that format, just that it exists and there's a literal datatype class for it. There are some notes about the WKT DGGS format in the content of this PR so they could be removed from GeoSPARQL and left up to the WKT DGGS formalisation to specify.

**Comment by rggibb** [comment]
From a DGGS perspective the suggested geometry literals are:
  **OrdinateList**, **DirectedOrdinateList**
  **CellList**, **DirectedCellList**
  **TileList**, **DirectedTileList**
  **ZoneCollection**, **DirectedZoneCollection**

In each of the above **Directed** is in the sense of a **Directed Graph**: So we will need to address how the Directed variants are encoded. There are lots of examples of Directed Graph implementations and maybe some of them are candidates for using in WKT. I’m thinking here in particular of directed graphs with lots of branching – as in a stream network. There will also be implementation issues that will need thinking through of how progressive detail (the sorting example above)  is applied to a complex branching directed graph

In addition to these we might want to consider options that can be applied to any of the above, such as:
a.	Compression – ie recursively replacing children by their parent, when all a parent’s children are present.
b.	Sorting – eg sort by increasing levels, which could be used in a streaming delivery to provide progressively greater detail 
c.	ASCII vs Binary – eg for those DGGS where the ZoneID is a binary code rather than an ASCII code, and the Binary form might be more compact.

**Comment by rggibb** [comment]
The descriptions of these DGGS geometry literals and the relationship between traditional Point, Line, Polygon geometries and the DGGS geometries are discussed in the OGC TB16 DGGS & DGGS API ER currently being published as 20-039. 

In summary the descriptions/preliminary definitions are:
**OrdinateList**: a list of (1..*) ZoneIds representing a set of points,
**DirectedOrdinateList**: a list of (2..*) ZoneIds ordered by connectivity representing a sequence of points, ZoneIds in the sequence need not touch, connectivity is assumed to be by ‘straight’ vectors,
**CellList**: a list of (1..*) ZoneIds representing a set of nD spaces
**DirectedCellList**: a list of (2..*) ZoneIds ordered by connectivity representing a sequence of nD spaces, all ZoneIds in the sequence need to touch at least one other connected ZoneId in the directed sequence, to ensure there are no gaps,
**TileList**: a list of (1..*) ZoneIds representing a set of nD tiles,
**DirectedTileList**: a list of (2..*) ZoneIds ordered by connectivity representing a sequence of nD spaces, ZoneIds in the sequence need not touch, connectivity is assumed to be by ‘straight’ vectors,
**ZoneCollection**: a list of (2..*) of one or more of the above types,
**DirectedZoneCollection**: a list of one or more of the above types that are connected.

Depending on the DGGS's dimensionality, these literals apply to all dimensionalities; 2D, 3D, 2D+T, 3D+T

**Comment by rggibb** [comment]
The Topic 21 v1.0 & v2.0 Abstract Specification defines quantization roles. In the following list
the quantization role is represented as a type cast from a traditional geometry literal to a DGGS Geometry literal. Best practice is to associate cell size with coordinate precision for asCoordinate.

- POINT.asCoordinates -> OrdinateList
- MULTIPOINT.asCoordinates -> OrdinateList
- LINESTRING.asCoordinates -> DirectedOrdinateList
- MULTILINESTRING.asCoordinates -> ZoneCollection of DirectedOrdinateList
- POLYGON.asCoordinates -> DirectedOrdinateList
- MULTIPOLYGON.asCoordinates -> DirectedZoneCollection of DirectedCoordinateLists
- TRIANGLE.asCoordinates -> DirectedOrdinateList
- TIN.asCoordinates -> DirectedZoneCollection of DirectedOrdinateList
- POINT.asDataCells -> CellList
- MULTIPOINT.asDataCells -> CellList
- LINESTRING.asDataCells -> DirectedCellList
- MULTILINESTRING.asDataCells -> ZoneCollection of DirectedCellList
- POLYGON.asDataCells -> CellList
- MULTIPOLYGON.asDataCells -> CellList
- TRIANGLE.asDataCells -> CellList
- TIN.asDataCells -> ZoneCollection of CellList
- GEOMETRYCOLLECTION.as<any> -> ZoneCollection

DGGS can represent geometries across a wider context than WKT geometry literals, so this next list illustrates the wider context:

- <point array>.asDataCells -> CellList
- <point cloud>.asDataCells -> CellList or 3D+T ZoneIds
- <raster image>.asCoordinates -> OrdinateList
- <raster image>.asDataCells -> CellList
- <raster image>.asDataTiles -> TileList
- <trajectory>.asCoordinates -> DirectedOrdinateList of 3D+T ZoneIds
- <trajectory>.asDataCells -> DirectedCellList of 3D+T ZoneIds
- <time series>.asCoordinates -> DirectedOrdinateList of 2D+T or 3D+T ZoneIds
- <time series>.asDataCells -> DirectedCellList of 2D+T or 3D+T ZoneIds
- <WMS image>.asGraphicCells -> CellList
- <WMS image collection>.asGraphicCells -> ZoneCollection of CellList
- <WMS image tile set>.asGraphicTiles -> TileList
- <Bounding-box>.asCoordinates -> OrdinateList
- <Bounding-box>.asDataTile -> TileList
- <Social media stream>.asTags -> DataCellList
- <Document text>.asTags -> DataCellList

File: 1.1/geo.ttl [hunk 1]
```diff
@@ -53,9 +53,30 @@
 	rdfs:label "Well-known Text Literal"@en ;
 	skos:definition """A Well-known Text serialization of a geometry object."""@en ;
 	skos:prefLabel "Well-known Text Literal"@en .
-	
+
 # 
-# http://www.opengis.net/ont/geosparql#wktLiteral
+# http://www.opengis.net/ont/geosparql#wktDGGSLiteral
+
+:wktDGGSLiteral a rdfs:Datatype ;
```
Comment (dr-shorthair) [inline 1.1/geo.ttl:60]: Please include a `skos:example`?

**Comment by mathib** [comment]
https://github.com/opengeospatial/ogc-geosparql/pull/48#issuecomment-756878955

**Comment by nicholascar** [comment]
I have re-issued the content of this, in updated form, in PR #88 so this PR will never be merged however I will leave it in Draft form (i.e. not closed) until PR #88 is merged due to the comments here.

**Comment by nicholascar** [comment]
Closing since the updated PR is merged

**Pull request closed without merging**